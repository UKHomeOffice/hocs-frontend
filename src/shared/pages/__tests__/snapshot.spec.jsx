import React from 'react';
const BASE_URL = '/api';
const MOCK_MATCH = { url: '/standard-lines' };

jest.mock('axios', () => ({
    get: jest.fn((url) => {
        switch (url) {
            case '/user/__TEST__/standardLine':
                return Promise.resolve({
                    data: [{
                        uuid: 'test',
                        displayName: 'test',
                        topicUUID: 'test',
                        expires: '9999-12-31',
                        documentUUID: 'test'
                    }]
                });
            default:
                return Promise.reject({
                    response: {
                        data: {}
                    }
                });
        }
    })
}));
jest.mock('../../contexts/actions/index.jsx', () => ({
    updateApiStatus: jest.fn(),
    clearApiStatus: jest.fn(),
}));

import axios from 'axios';
import WrappedStandardLines from '../standardLines/standardLinesView.jsx';
const FLUSH_PROMISES = () => new Promise(resolve => setImmediate(resolve));

describe('Workstack page component', () => {

    const MOCK_DISPATCH = jest.fn();

    beforeEach(() => {
        MOCK_DISPATCH.mockReset();
        MOCK_DISPATCH.mockReturnValue(Promise.resolve());
        axios.get.mockClear();
    });

    it('should render with default props', async () => {
        const OUTER = shallow(<WrappedStandardLines match={MOCK_MATCH}/>);
        const StandardLinesPage = OUTER.props().children;
        const WRAPPER = mount(
            <StandardLinesPage dispatch={MOCK_DISPATCH} />
        );
        await FLUSH_PROMISES();
        expect(WRAPPER).toBeDefined();
        expect(WRAPPER).toMatchSnapshot();
        expect(MOCK_DISPATCH).toHaveBeenCalled();
        expect(axios.get).toHaveBeenCalled();
        expect(axios.get).toHaveBeenCalledWith(BASE_URL + MOCK_MATCH.url);
    });
});
