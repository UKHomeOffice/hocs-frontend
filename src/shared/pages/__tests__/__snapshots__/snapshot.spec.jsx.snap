// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workstack page component should render with default props 1`] = `
<Component
  dispatch={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
        ],
        Array [
          undefined,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": Promise {},
        },
        Object {
          "type": "return",
          "value": Promise {},
        },
      ],
    }
  }
>
  <StandardLinesView
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      }
    }
    match={
      Object {
        "url": "/standard-lines",
      }
    }
  >
    <div
      className="govuk-form-group"
    >
      <div>
        <h1
          className="govuk-heading-xl"
        >
          View standard lines
        </h1>
        <div>
          <div
            className="govuk-grid-row"
          >
            <div
              className="govuk-grid-column-one-third"
            >
              <div
                className="govuk-form-group filter-row"
              >
                <Text
                  autoFocus={true}
                  disabled={false}
                  label="Filter"
                  limit={4000}
                  name="filter"
                  type="text"
                  updateState={[Function]}
                  value=""
                >
                  <div
                    className="govuk-form-group "
                  >
                    <label
                      className="govuk-label govuk-label--s"
                      htmlFor="filter"
                      id="filter-label"
                    >
                      Filter
                    </label>
                    <div>
                      <input
                        className="govuk-input"
                        disabled={false}
                        id="filter"
                        maxLength={4000}
                        name="filter"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </Text>
                <span
                  aria-live="polite"
                  className="govuk-hint"
                >
                  0 Items
                </span>
              </div>
              <div
                className="govuk-grid-row margin-bottom--small"
              >
                <div
                  className="govuk-grid-column-two-thirds govuk-label--s padding-top--small"
                >
                  Exclude expired
                </div>
                <div
                  className="govuk-grid-column-one-third bigger"
                >
                  <input
                    checked={true}
                    name="excludeExpired"
                    onChange={[Function]}
                    type="checkbox"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="scrollableTable"
          >
            <table
              className="govuk-table"
            >
              <thead
                className="govuk-table__head"
              >
                <tr
                  className="govuk-table__row"
                >
                  <th
                    className="govuk-table__header"
                    scope="col"
                  >
                    Topic
                  </th>
                  <th
                    className="govuk-table__header"
                    scope="col"
                  >
                    Team
                  </th>
                  <th
                    className="govuk-table__header"
                    scope="col"
                  >
                    Filename
                  </th>
                  <th
                    className="govuk-table__header"
                    scope="col"
                  >
                    Expiry date
                  </th>
                  <th
                    className="govuk-table__header"
                    scope="col"
                  >
                    Action
                  </th>
                </tr>
              </thead>
              <tbody
                className="govuk-table__body"
              />
            </table>
          </div>
        </div>
        <br />
      </div>
    </div>
  </StandardLinesView>
</Component>
`;
